network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,454 @@
+                            result = {'changed' : True, 'deleted' : name }
+                        except bigsuds.OperationFailed, e:
+                            if "was not found" in str(e):
+                                result['changed']= False
+                            else:
+                                raise
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state == 'present':
+            update = False
+            if not vs_exists(api, name):
+                if (not destination) or (not port):
+                    module.fail_json(msg="both destination and port must be supplied to create a VS")
+                if not module.check_mode:
+                    # a bit of a hack to handle concurrent runs of this module.
+                    # even though we've checked the pool doesn't exist,
+                    # it may exist by the time we run create_pool().
+                    # this catches the exception and does something smart
+                    # about it!
+                    try:
+                        vs_create(api,name,destination,port,pool)
+                        set_profiles(api,name,all_profiles)
+                        set_snat(api,name,snat)
+                        set_description(api,name,description)
+                        set_default_persistence_profiles(api,name,default_persistence_profile)
+                        result = {'changed': True}
+                    except bigsuds.OperationFailed, e:
+                            raise Exception('Error on creating Virtual Server : %s' % e)
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+            else:
+                update = True
+            if update:
+                # VS exists
+                if not module.check_mode:
+                    # Have a transaction for all the changes
+                    try:
+                        api.System.Session.start_transaction()
+                        result['changed']|=set_destination(api,name,fq_name(partition,destination))
+                        result['changed']|=set_port(api,name,port)
+                        result['changed']|=set_pool(api,name,pool)
+                        result['changed']|=set_description(api,name,description)
+                        result['changed']|=set_snat(api,name,snat)
+                        result['changed']|=set_profiles(api,name,all_profiles)
+                        result['changed']|=set_default_persistence_profiles(api,name,default_persistence_profile)
+                        api.System.Session.submit_transaction()
+                    except Exception,e:
+                        raise Exception("Error on updating Virtual Server : %s" % e)
+                else:
+                    # check-mode return value
+                    result = {'changed': True}
+
+        elif state in ('disabled', 'enabled'):
```

Comment:
Doesn't look like these states do anything. Do you want to add them? If not, we can remove them for now and patch them in later